test/penalizer/Penalizer.test.ts

```diff
@@ -290,6 +273,81 @@ contract('Penalizer', function ([relayOwner, relayWorker, relayManager, otherAcc
     })
   })
 
+  describe('should receive/reject claims according with the commitment time', function () {
+    before(async function () {
+      await penalizer.setHub(relayHub.address, { from: await penalizer.owner() })
+    })
+    beforeEach(async () => {
+      await token.mint('1000', forwarder.address)
+
+      await relayHub.stakeForAddress(relayManager, 1000, {
+        from: relayOwner,
+        value: ether('1'),
+        gasPrice: gasPrice
+      })
+      await fundAccount(fundedAccount, sender.address, '10')
+    })
+
+    // successful claims
+    const acceptableCommitmentTimes = [
+      // due to security implications, we accept a up to 15 secs of delay
+      // but we cannot set exactly this time in tests
+      10,
+      0,
+      // set a past commitment time
+      -100,
+      -150
+    ]
+    acceptableCommitmentTimes.forEach((timeDiff, index) => {
+      it(`accept them if commit time is ${Math.abs(timeDiff)} seconds in ${timeDiff < 0 ? 'past' : 'future'}`, async function () {
+        const [rr, sig] = await createRelayRequestAndSignature({ relayData: `0xdeadbeff1${index}`, enableQos: true })
+        const now = getSecondsSinceUnixEpoch()
+        const receiptTime = now + timeDiff
+        const receipt = relayHelper.createReceipt(rr, sig, receiptTime)
+        await relayHelper.signReceipt(receipt)
+
+        const rawTx = await createRawTx(
+          sender,
+          penalizer.address,
+          penalizer.contract.methods.claim(receipt).encodeABI(),
+          txGas.toString(),
+          gasPrice,
+          chainId
+        )
+        await assertClaimIsSuccessful(relayHub, relayManager, sender, rawTx)
+      })
+    })
+
+    // commitment time not yet expired, unsuccessful claims
+    // if we set a small future time, the claims can be successful
+    // due to the 15 seconds of delay we accept.
+    const futureReceiptTime = [
+      60,
```

Comment:
out of curiosity: how far is `60` to the case of an unexpected successful claim for this scenario?

i guess what i am asking is: do we have an idea of how much time drift are we working with in these cases?

(i suppose it could depend on the machine, but just to have a ballpark number)

Reply:
that's a legit question. I started using 50 as lower bound, but I noticed that it failed sometimes (not often, but it occurred), so I set a higher value to avoid that the tests fail unexpectedly.

Reply:
we could set lower values a take the "risk" that sometimes the tests fail. WDYT?

Reply:
> we could set lower values and take the "risk" that sometimes the tests fail. WDYT?

no, let's leave it as it is. i just did not expect to need 60 seconds, but i guess all of the test infrastructure (plus machine details) could justify this